meetings/2021-08-11.md

```diff
@@ -0,0 +1,126 @@
+# 2021-08-11 Solid Authorization
+
+https://meet.jit.si/solid-authorization
+
+
+## Agenda
+
+* Minutes 2021-07-28 - https://github.com/solid/authorization-panel/pull/245
+* Effective ACR discovery - https://github.com/solid/authorization-panel/pull/244
+
+
+## Present
+
+* Barath Raghavan
+* Sarven Capadisli
+* Henry Story
+* Elf Pavlik
+
+
+## Minutes
+
+### Welcome Barath
+
+Barath: There is some interesting material on https://lamport.azurewebsites.net/tla/tla.html.
+Someone used it to model an access control system https://arxiv.org/pdf/1806.09848.pdf.
+
+
+### Effective ACR
+
+Henry: If you have to use the default ACL, the client will have to make 1 request per container to go all the way to the default.
+
+Sarven: You can't pick a random number of depth in container hierarchy. You can have a simple solution where 1 access control is on the default directory and you have 1 level deep.
+
+Henry: I asked how it is meant to be done and this is what the process costs in terms of requests.
+
+Sarven: What is the criteria for fixing it? Is it the number of requests?
+
+Henry: Having a default is not a bad idea. It is a complicated process to get there.
+
+Sarven: To be clear, there is nothing that the WAC spec prohibits having each their ACR.
+
+Henry: I'm following the process that has to be followed and writing it to a wiki.
+It's not unworkable, it just has a lot of requests and is a bit inefficient.
+Do you disagree that this is inefficient or have I missed something?
+
+Sarven: I agree that this could be more efficient, but what are you going to adjust?
+
+Henry: My idea is to address it with the minimum changes to WAC.
+Here I can think of for example the rel link having a mime type `application/trig` which would be a quad representation. In which case, the server could return a graph with no content in the graph of the resource but tell you what the default graph is. With one get, you get the default ACL.
+
+Sarven: I think I'm a bit confused. There is a slight difference between the spec describing how the default works and allowing implementations to work their own implementations. 1 GET to the ACL would give you everything you need to know. As long as the server generates default authorization rules when a resource is created.
+The change that could be on the spec level would be if it prohibits that possibility from happening (ommit a representation from an ACL).
+The spec can say all references to an ACL must have a representation.
+Currently, that the thing we don't enforce.
+
+Henry: There are two answers: with the trig or n3 serialisation would allow us to have the graph of the resource marked empty and the graph of the default thrown in.
```

Comment:
```suggestion
Henry: There are two answers: either the trig or n3 serialisation would allow us to have the graph of the resource marked empty, and the graph of the default thrown in.
```